packages/arb-upgrades/index.ts

```diff
@@ -0,0 +1,361 @@
+import { HardhatRuntimeEnvironment } from 'hardhat/types/runtime'
+import { writeFileSync, readFileSync } from 'fs'
+import childProcess from 'child_process'
+import {
+  QueuedUpdates,
+  CurrentDeployments,
+  ContractNames,
+  CurrentDeployment,
+  QueuedUpdate,
+  isBeacon,
+} from './types'
+
+// import {
+//   getStorageLayout,
+//   assertUpgradeSafe,
+//   getContractVersion,
+//   assertStorageUpgradeSafe,
+//   solcInputOutputDecoder,
+//   validate,
+//   RunValidation,
+// } from '@openzeppelin/upgrades-core'
+
+const adminSlot =
+  '0xb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d6103'
+const implementationSlot =
+  '0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc'
+const beaconSlot =
+  '0xa3f0ad74e5423aebfd80d3ef4346578335a9a72aeaee59ff6cb3582b35133d50'
+
+const currentCommit = childProcess
+  .execSync('git rev-parse HEAD')
+  .toString()
+  .trim()
+
+const ensureCleanGitTree = () => {
+  const tree = childProcess.execSync('git diff-index HEAD --').toString().trim()
+  const found = tree.split('\n').find(diff => diff.includes('contracts'))
+  if (found) {
+    throw new Error(
+      `You have local changes to ${found}. Commit/stash/ get rid of them`
+    )
+  }
+}
+
+export const initUpgrades = (
+  hre: HardhatRuntimeEnvironment,
+  rootDir: string
+) => {
+  const getQueuedUpdates = async (): Promise<{
+    path: string
+    data: QueuedUpdates
+  }> => {
+    const network = await hre.ethers.provider.getNetwork()
+    const path = `${rootDir}/deployments/${network.chainId}_queued-updates.json`
+    try {
+      const jsonBuff = readFileSync(path)
+      return { path, data: JSON.parse(jsonBuff.toString()) as QueuedUpdates }
```

Comment:
I believe this is typesafe since we usually `writeFileSync(path, JSON.stringify(queuedUpdatesData))`
I'm not sure how this would behave if the QueuedUpdates json is manually changed to something incorrect